solc/CommandLineInterface.cpp

```diff
@@ -590,10 +590,13 @@ bool CommandLineInterface::readInputFilesAndConfigureRemappings()
 					continue;
 				}
 
-				m_sourceCodes[infile.generic_string()] = readFileAsString(infile.string());
+				m_sourceCodes[path] = readFileAsString(infile.string());
```

Comment:
Here we ensure the source mapping in `CompilerStack` is not tainted with the value of `--base-path`. This is only a change because above we prepend `m_basePath`.